File: app/src/main/java/ru/filchacov/billsplittest/FriendItem.java
package ru.filchacov.billsplittest;

import java.util.UUID;

public class FriendItem {
    private int mImageResource;
    private String mText1;
    private String mUid;

    public FriendItem(int mImageResource, String mText1, String mUid) {
        this.mImageResource = mImageResource;
        this.mText1 = mText1;
        this.mUid = mUid;
    }

    public int getmImageResource() {
        return mImageResource;
    }

    public String getmText1() {
        return mText1;
    }

    public String getmUid() {
        return mUid;
    }
}

File: app/src/main/java/ru/filchacov/billsplittest/AddFriend/FriendItem.kt
package ru.filchacov.billsplittest.AddFriend

import android.os.Parcel
import android.os.Parcelable

class FriendItem(private var mImageResource: Int, private var mText: String?): Parcelable{

    private var isSelected: Boolean = false

    private var key: String = ""

    constructor(parcel: Parcel) : this(
            parcel.readInt(),
            parcel.readString()) {
        isSelected = parcel.readByte() != 0.toByte()
        key = parcel.readString()!!
    }

    fun getKey(): String{
        return key
    }

    fun setKey(key: String){
        this.key = key
    }


    fun getisSelected(): Boolean{
        return isSelected
    }

    fun setisSelected(isSelected: Boolean){
        this.isSelected = isSelected
    }

    fun getmImageResource(): Int {
        return mImageResource
    }

    fun getmText(): String? {
        return mText
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(mImageResource)
        parcel.writeString(mText)
        parcel.writeByte(if (isSelected) 1 else 0)
        parcel.writeString(key)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<FriendItem> {
        override fun createFromParcel(parcel: Parcel): FriendItem {
            return FriendItem(parcel)
        }

        override fun newArray(size: Int): Array<FriendItem?> {
            return arrayOfNulls(size)
        }
    }


}